### PR TITLE
Collect Sensor Data and Export to the stackDriver

### PR DESCRIPTION
**Pull request by hbgxsm95**
For the main.go file, it collects sensor data including temperature, humidity, sound and light sensor data and export them to the stackDriver.  For the sound sensor data, I use two aggregation method to process the raw data in order to compare the difference of them. 

For the compare.go file, it collects temperature sensor data. It's from another sensor but with same model and connected to the different pin on the raspberry Pi.  The main and compare program should run concurrently. In this way, we could compare the performance including precision and latency of the sensors with the same models..

### CODE REVIEW

File: go/iot/main.go [hunk 1]
```diff
@@ -93,22 +200,68 @@ func initOpenCensus(projectId string, reportPeriod int) {
...
 
-	// Set reporting period to report data at every second.
+	// Set reporting period to report data based on the given reportPeriod.
 	view.SetReportingPeriod(time.Second * time.Duration(reportPeriod))
 
-	// Create view to see the processed video size cumulatively.
+	// Create view to see the sound strength distribution.
+	// Subscribe will allow view data to be exported.
+	// Once no longer need, you can unsubscribe from the view.
+	if err := view.Register(&view.View{
```
Comment (songy23) [inline go/iot/main.go]: nit: consider defining these views as constants, similar to soundStrengthDistMeasure, etc.

Comment (hbgxsm95) [reply go/iot/main.go]: Sounds great

File: go/iot/main.go [hunk 2]
```diff
@@ -93,22 +206,63 @@ func initOpenCensus(projectId string, reportPeriod int) {
...
+	// Subscribe will allow view data to be exported.
+	// Once no longer need, you can unsubscribe from the view.
+	if err := view.Register(viewToRegister); err != nil {
+		log.Fatalf("Cannot subscribe to the view: %v", err)
+	}
+
+	// Create view to see the sound strength instantly.
+	// Subscribe will allow view data to be exported.
+	// Once no longer need, you can unsubscribe from the view.
+	if err := view.Register(&view.View{
```
Comment (songy23) [inline go/iot/main.go]: Same for this one. You can also make a slice of views that need registering, and register them in a loop.

File: go/iot/main.go [hunk 3]
```diff
@@ -93,22 +206,63 @@ func initOpenCensus(projectId string, reportPeriod int) {
...
+	}); err != nil {
+		log.Fatalf("Cannot subscribe to the view: %v", err)
+	}
+
+	// Create view to see the light strength instantly.
 	// Subscribe will allow view data to be exported.
 	// Once no longer need, you can unsubscribe from the view.
 	if err := view.Register(&view.View{
-		Name:        "my.org/views/gpio_voltage_instant",
+		Name:        "opencensus.io/views/light_strength_instant",
```
Comment (songy23) [inline go/iot/main.go]: ditto

File: go/iot/main.go [hunk 4]
```diff
@@ -93,22 +206,63 @@ func initOpenCensus(projectId string, reportPeriod int) {
...
+		Measure:     lightStrengthMeasure,
+		Aggregation: view.LastValue(),
+	}); err != nil {
+		log.Fatalf("Cannot subscribe to the view: %v", err)
+	}
+
+	// Create view to see the humidity instantly.
+	// Subscribe will allow view data to be exported.
+	// Once no longer need, you can unsubscribe from the view.
+	if err := view.Register(&view.View{
```
Comment (songy23) [inline go/iot/main.go]: ditto

File: go/iot/main.go [hunk 5]
```diff
@@ -93,22 +206,63 @@ func initOpenCensus(projectId string, reportPeriod int) {
...
 		log.Fatalf("Cannot subscribe to the view: %v", err)
 	}
-}
 
-func recordVoltage(ctx context.Context, voltage int64) {
-	stats.Record(ctx, gpioVoltage.M(voltage))
+	// Create view to see the temperature instantly.
+	// Subscribe will allow view data to be exported.
+	// Once no longer need, you can unsubscribe from the view.
+	if err := view.Register(&view.View{
```
Comment (songy23) [inline go/iot/main.go]: ditto

**Pull request merged**